scripts/snapshot-restore-balance/snapshot-balances.js

```diff
@@ -0,0 +1,81 @@
+const { ApiPromise, WsProvider, Keyring } = require('@polkadot/api');
+const fs = require('fs');
+
+const WS_ENDPOINT = 'wss://testnet-rpc.zkverify.io';
+const DISCARDED_ACCOUNTS = [
+        "5D9txxK9DTvgCznTjJo7q1cxAgmWa83CzHvcz8zhBtLgaLBV",
+        "5ETuZEyLnfVzQCaDM8aQCcsNnz6xjPKvQCtqynCLqwng8QLd",
+        "5D29UEzgStCBTnjKNdkurDNvd3FHePHgTkPEUvjXYvg3brJj",
+        "5DiMVAp8WmFyWAwaTwAr7sU4K3brXcgNCBDbHoBWj3M46PiP",
+        "5EhREncHsntgJaax9YQphk1xN3LxPu2Rzbz4A3g7Ut8cRXWq",
+        "5C84NU2477uHCUF1A8rHb89sP2D2ZsnquPaGa2Htv75FN9gm",
+        "5HdZjrmNAkWQhYQUPNv7YRYnT4vyQswjbNm8eXBvULNQz5wH",
+        "5HjFLKpiCStQgRm6ZM1fT1R5pLKAqQdUG3uh7pvzaQfhdFuB",
+        "5ECktCamcAtBFJirEzvvJmXFxgLMCTAejhqZwLT1Dxn2fwB1",
+        "5EZbvFqx3j7ejqBSPWseif8xL3PwoqMQHdMT8rs9qWoHcdR3"
+    ]
+
+async function takeSnapshot(api, snapshotBalancesFile) {
+    const balances = {};
+    const accounts = await api.query.system.account.keys();
+
+    for (const key of accounts) {
+        const account = key.args[0].toString();
+        if (DISCARDED_ACCOUNTS.includes(account)) continue;
+        const { data: { free: balance } } = await api.query.system.account(account);
```

Comment:
I might be missing some context, but I believe that we should also consider the `reserved` balance here.
For example, `reserved` balance should be transformed back into `free` balance if the operations that the balanced was reserved for are lost in the regenesis.